### ISSUE TITLE
Support for dynamic alias names in dbt

### ISSUE THREAD

**Issue by btello** [datafold/data-diff#529]
**Is your feature request related to a problem? Please describe.**
In our environment, we leverage dynamic alias names using dbt's `generate_alias_name` macro:

https://docs.getdbt.com/docs/build/custom-aliases#generate_alias_name

For example, let's say we have a model named `foo`.  In `dev` the model gets aliased to `stage__foo` and in production the alias is just `foo`. 

A good breakdown of why we implemented things this way as well as the actual code we're using in `generate_alias_name` can be found on this blog post:

https://medium.com/data-manypets/how-to-customise-dbts-model-naming-for-easy-developing-on-production-1238559a939a

The problem is that when I run `data-diff --dbt` it uses the `dev` version of the alias for both the dev and prod targets and ultimately this means I can't use data-diff in my environment (which I very much want to do :-) ).

Example:

```
> data-diff --dbt
Running with data-diff=0.7.4

dbxyz.stage.stage__foo <> dbxyz.dbt_dev.stage__foo
New model or no access to prod table.
```

**Describe the solution you'd like**

I would like to be able to use data-diff in an environment where I have customized dynamic alias names as described above.

**Describe alternatives you've considered**

I could add a static custom alias to each of my models but I've got several hundred models and this solution would just be very error prone.

**Additional context**
None

**Comment by followingell** [comment]
+1 for this

**Comment by dlawin** [comment]
Hi @btello, there is support for dynamic model names now via the `--state` flag. We could also add support via configuration, but it's unlikely to be a big priority now that `--state` is available.

https://docs.datafold.com/development_testing/open_source#configure-your-dbt-project
![Screenshot 2023-07-05 at 10 22 48 AM](https://github.com/datafold/data-diff/assets/11282254/dc7b2c7d-a356-4d53-b515-bf22c183998f)

**Comment by followingell** [comment]
@dlawin Thanks for the reply!

After testing `--state` doesn't work for my implemetaion, details below 🙂 
For our `dev` target we utilise the `{{ node.name }}` _(AKA model file name)_ as the alias and for the `prod` target we use the `custom_alias_name`.

Is there a way that I can compare the same model across `dev` and `prod` given the above setup? I believe the package takes the alias and applies it to both targets regardless of actual database behaviour:
- **Actual `dev` target database identifier:** `dev_catalog.specific_user_dev_schema.alias_plus_node_name`.
- **Actual `prod` target database identifier:** `prod_catalog.project_schema.node_name`

- **Datafold `dev` identifier:** `dev_catalog.specific_user_dev_schema.alias_plus_node_name` :white_check_mark:
- **Datafold `prod` identifier:** `prod_catalog.project_schema.alias_plus_node_name` :x:

**Comment by dlawin** [comment]
Hey @btello @followingell this should be resolved in versions >=0.8.0 via use of the `--state` flag (we were inadvertently using the alias from the "dev" manifest) 

Duplicate issue: https://github.com/datafold/data-diff/issues/646

Can you reopen this issue if there are any remaining problems?

**Comment by followingell** [comment]
@dlawin Finally got around to testing this and it's working great, thank you!
